Add section about validating Saleor webhook signature (#464)
saleor/saleor-docs: docs/developer/extending/apps/asynchronous-webhooks.mdx

- if `secretKey` was set for a webhook - the HMAC SHA-256 header based on it and the payload
- if `secretKey` wasn't set for a webhook - the JWS signature using RS256 with payload detached, to verify signature you can use public key, which can be fetched from `http(s)://<your-backend-domain>/.well-known/jwks.json` (available since Saleor 3.5, before 3.5 payload from webhooks without `secretKey` set were not signed)

##### Validating signature

> To validate signatures in Saleor Apps use [`withWebhookSignatureVerified`](https://github.com/saleor/saleor-app-sdk/blob/main/src/middleware.ts#L62) middleware provbided by [`saleor-app-sdk`](https://github.com/saleor/saleor-app-sdk)

Reviewer: @witoszekdev typo `provbided`

```js
import type { Handler } from "retes";
import { Response } from "retes/response";
import { toNextHandler } from "retes/adapter";
import {
  withSaleorEventMatch,
  withWebhookSignatureVerified,
} from "@saleor/app-sdk/middleware";

const handler: Handler = async (request) => {
  // ...

  return Response.OK({ success: true });
};

export default toNextHandler([
  withSaleorDomainMatch,
  withWebhookSignatureVerified(),
  handler,
]);
```

> Example taken from [`saleor-app-template`](https://github.com/saleor/saleor-app-template/blob/main/pages/api/webhooks/order-created.ts)

> You can also manually validate the JWS signature in Javascript with [`jose` package](https://github.com/panva/jose). Remember that you need to get raw body string, not parsed by some middleware

Reviewer 2: ```suggestion
You can also manually validate the JWS signature in Javascript with [`jose` package](https://github.com/panva/jose). Remember that you need to supply it with a raw body string, not a parsed one.
```

@witoszekdev please confirm it still makes sense.

Author: ```suggestion
You can also manually validate the JWS signature in Javascript with [`jose` package](https://github.com/panva/jose). Remember that you need to supply it with a raw body string, not a parsed object.
```

I think this would make more sense - what I meant here is that you need to pass a string of body, not `JSON.parse`'d  by some middleware. For example Next.js does this by default and you need to explicity disable bodyParser middleware, like this:

```ts
export const config = {
  api: {
    bodyParser: false,
  },
}
```

Maybe we could actually link to Next.js docs but I'm not sure about this - since the docs are written pretty much framework or language agnostic.

> You can also manually validate the JWS signature in Javascript with [`jose` package](https://github.com/panva/jose). Remember that you need to supply it with a raw body string, not a parsed object.

Reviewer 2: ```suggestion
You can also manually validate the JWS signature in JavaScript with [`jose` package](https://github.com/panva/jose). Remember that you need to supply it with a raw body string, not a parsed object.
```
